app/controllers/orders_controller.rb

```diff
@@ -259,6 +259,9 @@ def purchase
     end
 
     if order_purchaser.success?
+      should_show_admin_hold_warning = @order.order_details.map(&:reservation).compact.map(&:conflicting_admin_reservation).any?
```

Comment:
Does `@order.order_details.map(&:reservation).compact.any?(&:conflicting_admin_reservation)` work? It's a minor thing, but that would be faster in the average case because it would stop looking up conflicts as soon as it finds one.